stock_dashboard_api/utils/pool.py

```diff
@@ -0,0 +1,55 @@
+import logging
+import os
+import time
+from logging.config import fileConfig
+from pathlib import Path
+
+from psycopg2.pool import PoolError, SimpleConnectionPool
+
+POOL_DELAY = os.getenv('POOL_DELAY')
+
+fileConfig((Path.cwd().parent / 'logging.conf'), disable_existing_loggers=True)
+logger = logging.getLogger('pool')
+
+
+class Connection:
+    connection_pool = None
+
+    def __init__(self):
+        if not Connection.connection_pool:
+            Connection.connection_pool = SimpleConnectionPool(int(os.getenv('MINCONN')),
+                                                              int(os.getenv('MAXCONN')),
+                                                              user=os.getenv('POSTGRES_USER'),
+                                                              password=os.getenv('POSTGRES_PASSWORD'),
+                                                              host=os.getenv('POSTGRES_HOST'),
+                                                              port=os.getenv('POSTGRES_PORT'),
+                                                              database=os.getenv('POSTGRES_DB'))
+
+        logger.info('Connection pool was created')
+        self.conn = None
+        self.cursor = None
+
+    def __enter__(self):
+        logger.info(f'Get connection from pool {id(self.conn)}')
+        try:
+            self.conn = Connection.connection_pool.getconn()
+            self.conn.autocommit = False
+            self.cursor = self.conn.cursor()
+            return self
+
+        except PoolError:
+            logger.info('Pool doesn\'t have available connection. Please wait')
+            time.sleep(int(POOL_DELAY))
```

Comment:
What is the reason of this sleep?
In case `except PoolError` this will log info message, do some sleep stuff and return nothing.
Add retry logic please.